sdk-lib/src/main/java/cash/z/ecc/android/sdk/Synchronizer.kt

```diff
@@ -805,8 +824,7 @@ interface Synchronizer {
             val chainTip =
                 when (walletInitMode) {
                     is RestoreWallet -> {
-                        // TODO [#1772]: redirect to correct service mode after 2.1 release
-                        when (val response = downloader.getLatestBlockHeight(ServiceMode.Direct)) {
+                        when (val response = downloader.getLatestBlockHeight(ServiceMode.UniqueTor)) {
```

Comment:
This needs to switch on `sdkFlags.isTorEnabled`, otherwise it *will* (if the rest of the existing impl is correct) fail if `torClient == null` (because if Tor is supposed to be enabled that would correctly be an error), when we actually want this to succeed if Tor is supposed to be disabled.